src/components/Api.js

```diff
@@ -0,0 +1,84 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
```

Comment:
Needs correction: I can't edit the profile. I see the following payload of the request in the network tab:

```js
{
  "avatar": "TypeError: Failed to fetch     at Api.addCard (Api.js:40:12)     at PopupWithForm.handleAddCardSubmit [as _handleFormSubmit] (index.js:63:6)     at HTMLFormElement.<anonymous> (PopupWithForm.js:32:12)"
}
```